### PR TITLE
fix(stylish): support hint and print proper severity

### PR DESCRIPTION
**Pull request by P0lip**
Fixes https://github.com/stoplightio/spectral/issues/565

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Screenshots**

After:
image
image
image
image

### CODE REVIEW

File: src/formatters/__tests__/__fixtures__/oas3-schema-errors.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 [
   {
     "code": "oas3-schema",
-    "summary": "Validate structure of OpenAPIv3 specification.",
```
Comment (marbemac) [inline src/formatters/__tests__/__fixtures__/oas3-schema-errors.json]: Why did this go away? Was summary deprecated, and this PR is removing it?

File: test-harness/scenarios/results-skip-rule.oas3.scenario [hunk 2]
```diff
@@ -15,4 +15,4 @@ OpenAPI 3.x detected
  1:1  warning  api-servers       OpenAPI `servers` must be present and non-empty array.
  2:6  warning  info-description  OpenAPI object info `description` must be present and non-empty string.
 
-✖ 2 problems (0 errors, 2 warnings, 0 infos)
\ No newline at end of file
+✖ 2 problems (0 errors, 2 warnings, 0 infos, 0 hints)
```
Comment (marbemac) [inline test-harness/scenarios/results-skip-rule.oas3.scenario]: Should we just tackle proper pluralization for the values in this message in this PR? Suggest either:

- https://github.com/prantlf/fast-plural-rules
- https://github.com/blakeembrey/pluralize

File: src/formatters/__tests__/__fixtures__/oas3-schema-errors.json (hunk 1 above)
Comment (P0lip) [reply src/formatters/__tests__/__fixtures__/oas3-schema-errors.json]: >Why did this go away? Was summary deprecated, and this PR is removing it?

We removed `summary` some time ago. The fixture was just not updated at the time we did it.

File: test-harness/scenarios/results-skip-rule.oas3.scenario (hunk 2 above)
Comment (P0lip) [reply test-harness/scenarios/results-skip-rule.oas3.scenario]: We already have a simple util for that.
```ts
function pluralize(word: string, count: number): string {
  return count === 1 ? word : `${word}s`;
}
```
I guess this should be quite enough for the simple case we deal with?

Comment (marbemac) [reply test-harness/scenarios/results-skip-rule.oas3.scenario]: Yeah that sounds good to me!

**Pull request merged**